board_generator.pl

```diff
@@ -8,6 +14,46 @@
     generate_list(1, MatrixSize, -1, LastRow),
     long_conc([[FirstRow], BlackRows, EmptyRows, WhiteRows, [LastRow]], InitialBoard).
 
+% generate board after a left / right move
+generate_changed_board(BoardState, FirstChangedSlotRow, FirstChangedSlotCol, direction(0, Y), NextSlotsSequence, NextBoardState):-
+    board_size(BoardSize),
+    % Get before & after rows
+    BeforeRowIndex is FirstChangedSlotRow - 1,
+    AfterRowIndex is FirstChangedSlotRow + 1,
+    BottomBorderIndex is BoardSize + 1,
+    findall(
+        Row, 
+        (row(BoardState, RowIndex, Row), between(0, BeforeRowIndex, RowIndex)), 
+        BeforeRows
+    ),
+    findall(
+        Row, 
+        (row(BoardState, RowIndex, Row), between(AfterRowIndex, BottomBorderIndex, RowIndex)), 
+        AfterRows
+    ),
+
+    % Generate changed row
+    slots_sequence_by_direction(BoardState, BoardSize, FirstChangedSlotRow, FirstChangedSlotCol, direction(0, Y), CurrSlotTypes),
+    row(BoardState, FirstChangedSlotRow, ChangedRow),
+    ((
+        Y > 0, !,
+        append([UnchangedSlotTypes, CurrSlotTypes], ChangedRow), !,
+        append([UnchangedSlotTypes, NextSlotsSequence], NextEffectedRow)
+    )
+    ;
+    (
+        reverse(NextSlotsSequence, OrientedNextSlotsSequence),
```

Comment:
Oriented -> Reversed?

Reply:
I write oriented when I mean that slot sequence is oriented to board's rows and columns directions.
In the code I'm writing now that allows vertical moves, I'm doing a similar thing.
I'll show you and then we can talk about it, but I think the name `reversed` does not explain why I reversed the list